otherlibs/stdune-unstable/proc.ml

```diff
@@ -7,7 +7,6 @@ let restore_cwd_and_execve prog argv ~env =
       Unix.create_process_env prog argv env Unix.stdin Unix.stdout Unix.stderr
     in
     match snd (Unix.waitpid [] pid) with
-    | WEXITED 0 -> ()
```

Comment:
I don't really understand this change: is it related to the goal of this PR, or is it a drive-by fix?

Reply:
Basically, we were doing a call to `Proc.restore_cwd_and_execve` in a tail position in `bin/exec.ml`. After spreading the monad, this tail position is now expected to have type `unit Fiber.t`. Because the return type of `Proc.restore_cwd_and_execve` is `unit`, it didn't work. The simple fix is to add a `Fiber.return ()` after the call to `Proc.restore_cwd_and_execve`, but this is odd because this point is unreachable. Another way is to make `Proc.restore_cwd_and_execve` return `_`, which is what `Unix.execve` does. To do that, I had to replace this `()` by `exit 0`. 

In any case, this change is good because it aligns more the behaviour of `Proc.restore_cwd_and_execve` between windows and linux.

Reply:
Now that I say this, I realise that we should be calling `sys_exit` rather than `exit`. I updated the code and added a comment.

Reply:
Got it, thanks for the explanation!